pkg/build/builder/docker.go

```diff
@@ -94,6 +101,12 @@ func (d *DockerBuilder) Build() error {
 
 	cname := containerName("docker", d.build.Name, d.build.Namespace, "post-commit")
 	if err := execPostCommitHook(d.dockerClient, d.build.Spec.PostCommit, buildTag, cname); err != nil {
+		updateBuildFailureReason(d.build, api.StatusReasonPostCommitHookFailed)
+		if _, updateErr := d.client.UpdateDetails(d.build); updateErr != nil {
+			glog.V(0).Infof("error: An error occured while updating the build status %v", updateErr)
+			// NOTE: handle me!
```

Comment:
this validation https://github.com/openshift/origin/pull/10817/files#diff-a34690b2fc7db648fee44d96008026e0R128
but nvm about that i made a mistake :) my bad.

on the event recorder,  i came to the conclusion that it's a bit more complicated than i thought. 
in order to attach a new recorder i need to do something along the lines of this https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L79-L94 which is basically instantiate a new eventbroadcaster with a kubeclient interface.
while that works in factory, it's increasingly difficult to do in the builders because i have to re-instantiate everything, (especially the kubeclient https://github.com/openshift/origin/blob/master/pkg/build/controller/factory/factory.go#L80) which i think leads me astray from the purpose of the pr and unless we really, really want to add event recorders for the builders to add more debug output, it's not much added value right now just to show that we failed to update the build object, besides an obvious error.
on the other hand, it seems we generally use events to show the user what the pod was through, rather than show any builder events so this would sort of break that pattern, for what i've seen in parallel with what we do with deployments as well. (i didn't look that in depth, though)

i was wondering if there's anyway to reuse the buildcontroller's event recorder (https://github.com/openshift/origin/blob/master/pkg/build/controller/controller.go#L30) so that i'd have to avoid re-instantiating this. otherwise drop this and just show the error and leave this for a follow up PR.
unless you know some other way?